code/game/gamemodes/miniantags/hades/hades.dm

```diff
@@ -0,0 +1,247 @@
+				Appear()
+				var/sinPersonchoice = pick(list("Greed","Gluttony","Pride","Lust","Envy","Sloth","Wrath"))
+				switch(sinPersonchoice)
+					if("Greed")
+						src.say("Your sin, [sinPerson], is Greed.")
+						if(prob(50))
+							src.say("I will indulge your sin, [sinPerson].")
+							var/list/greed = list(/obj/item/stack/sheet/mineral/gold,/obj/item/stack/sheet/mineral/silver,/obj/item/stack/sheet/mineral/diamond)
+							for(var/i in 1 to 10)
+								var/greed_type = pick(greed)
+								new greed_type(get_turf(sinPerson))
+						else
+							src.say("Your sin will be punished, [sinPerson]!")
+							new/obj/structure/closet/statue(get_turf(sinPerson),sinPerson)
+					if("Gluttony")
+						src.say("Your sin, [sinPerson], is Gluttony.")
+						if(prob(50))
+							src.say("I will indulge your sin, [sinPerson].")
+							var/list/allTypes = list()
+							for(var/A in typesof(/obj/item/weapon/reagent_containers/food/snacks))
+								var/obj/item/weapon/reagent_containers/food/snacks/O = A
+								if(initial(O.cooked_type))
+									allTypes += A
+							for(var/i in 1 to 10)
+								var/greed_type = pick(allTypes)
+								new greed_type(get_turf(sinPerson))
+						else
+							src.say("Your sin will be punished, [sinPerson]!")
+							sinPerson.reagents.add_reagent("nutriment",1000)
+					if("Pride")
+						src.say("Your sin, [sinPerson], is Pride.")
+						if(prob(50))
+							src.say("I will indulge your sin, [sinPerson].")
+							var/obj/item/weapon/twohanded/sin_pride/good = new/obj/item/weapon/twohanded/sin_pride(get_turf(sinPerson))
+							good.pride_direction = 0
+						else
+							src.say("Your sin will be punished, [sinPerson]!")
+							var/obj/item/weapon/twohanded/sin_pride/bad = new/obj/item/weapon/twohanded/sin_pride(get_turf(sinPerson))
+							bad.pride_direction = 1
+					if("Lust")
+						src.say("Your sin, [sinPerson], is Lust.")
+						if(prob(50))
+							src.say("I will indulge your sin, [sinPerson].")
+							var/obj/item/lovestone/good = new/obj/item/lovestone(get_turf(sinPerson))
+							good.lust_direction = 0
+						else
+							src.say("Your sin will be punished, [sinPerson]!")
+							var/obj/item/lovestone/bad = new/obj/item/lovestone(get_turf(sinPerson))
+							bad.lust_direction = 1
+					if("Envy")
+						src.say("Your sin, [sinPerson], is Envy.")
+						if(prob(50))
+							src.say("I will indulge your sin, [sinPerson].")
+							for(var/mob/living/carbon/human/H in player_list) // name lottery
+								if(H == sinPerson)
+									continue
+								if(prob(25))
+									spawn(10)
+										sinPerson.name = H.name
+										sinPerson.real_name = H.real_name
+										var/datum/dna/lottery = H.dna
+										lottery.transfer_identity(sinPerson, transfer_SE=1)
+										sinPerson.updateappearance(mutcolor_update=1)
+										sinPerson.domutcheck()
+						else
+							src.say("Your sin will be punished, [sinPerson]!")
+							var/sinPersonspecies = pick(species_list)

```

Comment:
To be Honest @Shadowlight213 in a large set of cases it'll be bad, so I don't think a small chance of it being not-so-bad would be terrible.